internal/runner/operation.go

```diff
@@ -225,8 +226,19 @@ func (r *Runner) executeJob(
 		}
 	}
 
-	// evaluate all variables against the variable blocks we just decoded
-	inputVars, diags := variables.EvaluateVariables(log, pbVars, cfg.InputVariables)
+	// Evaluate all variables against the variable blocks we just decoded:
+	// We grab the server cookie here to pass along for the variables
+	// evaluation to use a salt for sensitive values
+	clientResp, _ := r.client.GetServerConfig(context.Background(), &empty.Empty{})
+	var serverCookie string
+	if clientResp != nil && clientResp.Config != nil {
+		serverCookie = clientResp.Config.Cookie
+	}
+	// We set both inputVars and jobVars on the project.
+	// inputVars is the set of cty.Values to use in our hcl evaluation
+	// and jobVars is the matching set of variable refs to store on the job that
+	// has sensitive values obfuscated and is used for user-facing feedback/output.
+	inputVars, jobVars, diags := variables.EvaluateVariables(log, pbVars, cfg.InputVariables, serverCookie)
```

Comment:
Pairing with the next comment, if you remove refs from operations, I'd update the job here with the rendered variables. This includes updating the server that the variables are now available... this might require a new event.